packages/itinerary-body/src/TransitLegBody/index.tsx

```diff
@@ -118,8 +120,13 @@ class TransitLegBody extends Component<Props, State> {
         ? transitOperator.logo
         : agencyBrandingUrl;
 
-    const expandAlerts =
-      alertsExpanded || (leg.alerts && leg.alerts.length < 3);
+    const maximumAlertCountToShowUncollapsed = 2;
+    const shouldCollapseDueToAlertCount =
+      leg.alerts?.length > maximumAlertCountToShowUncollapsed;
+    const shouldOnlyShowAlertsExpanded =
+      !(shouldCollapseDueToAlertCount || shouldAlwaysCollapseAlerts) ||
+      !leg.alerts;
```

Comment:
Is expanding alerts relevant (or desired) if `leg.alerts` is `null`?

Reply:
Yes, in fact it's needed - if `shouldOnlyShowAlertsExpanded` is false, we show the alert toggle. Can rename the variable (or leave a comment) if you'd like that to be a bit clearer.

Reply:
A comment about expanding alerts even if `leg.alerts` is `null` would be useful, please.

Reply:
Good call, thanks. I've added a comment. Please let me know if you find it unclear, and want tweaks to it.